lib/spack/spack/cmd/include.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.environment as ev
+import spack.paths
+from spack.util.path import substitute_path_variables
+
+description = "manage included configs and concrete environments in the active environment"
+section = "environments"
+level = "long"
+
+subcommands = ["add", "remove", "list"]
+
+
+def include_list_setup_parser(subparser):
+    """List include(s) in the active environment"""
+
+    expand_group = subparser.add_mutually_exclusive_group()
+    expand_group.add_argument("--expand", action="store_true")
+    expand_group.add_argument("--no-expand", action="store_false", dest="expand")
+
+    # TODO: Add recursive listing option for environments including environments with include
+    #       The default for list should be to only list the comonents listed in the active
+    #       environment.
+
+
+def include_list(env, args):
+    include = env.manifest.list_includes(concrete=args.concrete)
+
+    if sys.stdout.isatty():
+        if include:
+            tty.msg(
+                "Included "
+                + ("Concrete Environments" if args.concrete else "Configuration Scopes")
+            )
+        else:
+            tty.msg("No includes found")
+
+    if include:
+        msg = ""
+        for inc in include:
+            if args.expand:
+                formatted_inc = substitute_path_variables(inc)
+            else:
+                # Collapse the prefix to the active environment or spack root variables
+                formatted_inc = inc.replace(env.path, "$env")
+                formatted_inc = formatted_inc.replace(spack.paths.spack_root, "$spack")
+
+            # Print the env name and path
+            env_name = ev.environment_name(inc)
+            if not env_name == inc:
+                formatted_inc = f"{env_name} ({formatted_inc})"
+
+            msg += f"\n\t{formatted_inc}"
+
+        tty.msg(msg)
+
+
+def include_add_setup_parser(subparser):
+    """Add include(s) to the active environment"""
+    subparser.add_argument(
+        "includes",
+        metavar="includes",
+        nargs="+",
+        help="List of path(s) to configs or environments to include",
+    )
+
+
+def include_add(env, args):
+    with env.write_transaction():
+        invalid = env.manifest.add_includes(args.includes, concrete=args.concrete)
```

Comment:
It would be handy to support names of Spack-managed environments (e.g., converted to the  path using spack.environment.env_root_path()).